decisions/DR-update-devkit-architecture.md

```diff
@@ -0,0 +1,63 @@
+That puts a burden on the implementing system to
+- make sure every data point collected by the system is as well referenced (metadata) in the Privacy Computation Engine database
+- collect or transform data to a particular format (capture and fragments)
+
+This DR proposes removing the notion of Data Captures and Capture Fragments from Privacy Computation Engine.
+
+## Proposal
+
+The [Privacy computation engine](https://github.com/blindnet-io/product-management/tree/main/refs/high-level-architecture#privacy-computation-engine) (PCE) should have the following components:
+- Privacy Compiler
+- Privacy Request Capture Interface
+- Data Consumer/DPO Interface
+- Data access engine
+- Data storage connectors (for e.g. Postgres, MongoDB, Azure Storage)
+
+**Data access engine** is a component deployed by a client. It communicates with the **Privacy Compiler** and a **Data storage connector**.
+
+A **Data storage connector** is a component that has mappings that queries the data storage. It can be implemented by blindnet or by a client for an unsupported storage or if the data model is not suitable.
+It should containg a set of rules (mappings) on how to map a `selector` to a corresponding fetch query (`selector` to a e.g. table column in a relational database). 
+
+PCE is [configurable](https://github.com/blindnet-io/product-management/blob/2645056f4477a93b5a6635129c77a3e7fe4d9a9b/refs/schemas/priv/expected-behavior.md#configuration-and-prerequisites) with the following:
+- Parameters for `TRANSPARENCY` requests
+- `selectors`
+  - retention policies for every `selector`
+  - `privacy scope` for every `selector`
+- Legal bases with privacy scopes
+  - necessary
+  - contracts
+  - legitimate interests
+  - consent templates (default consents)
+- Regulations
+
+Regulation is defined as `(privacy scope)-(legal base)` which means data in `privacy scope` can only be processed if a user has the active `legal base` (e.g. has given consent).
+
+A new user can be created in the PCE in the following ways:
+- manually, with a request
+- upon giving a consent
+- upon making a privacy request
+
+By default, the user's eligible privacy scope is composed of privacy scope triples from `necessary`, `contracts` and `legitimate interests` legal bases, with regards to all `Regulations`. Every subsequent consent or privacy request will change the user's eligible privacy scope.
+
+### TBD Retention policy
+
+Since metadata can't be accessed by a privacy compiler, handling retention policies becomes a tricky problem.
+Data access engine can handle them by querying the storage on fixed intervals. That might put burden of additional development on a client.
+
```

Comment:
Can't the client also provide some sort of mappign between Retention Policy Events, and storage data/matadata so that PCE can generate queries and to check for expired data?

Reply:
That's how I imagined it to work.

Reply:
so what is the problem then?

Reply:
it's not a problem really, only it's more complicated than it used to be when the storage was handled by the devkit. the mapping (data access / storage connector) needs to handle relatively complex queries (not just "get this data with this ID", but also "all the data matching this expiry criteria"), that's it

Reply:
> the mapping (data access / storage connector) needs to handle relatively complex queries (not just "get this data with this ID", but also "all the data matching this expiry criteria")

It's obvious to me that when working with relational databases, this _mapping (data access / storage connector)_ is more complex then when we have a capture/fragment-aware storage. The clients adopting our storage will benefit from a more elegant (and maybe easier to configure) solution. That's ok.

The key requirement here is that the PCE MUST be agnostic to the type of storage used. It communicates with an interface general enough to abstract both this and the future storage we'll make (with captures and fragements)